packages/sdk/src/evm/core/sdk.ts

```diff
@@ -549,21 +550,38 @@ export class ThirdwebSDK extends RPCConnectionHandler {
 
   public async getMultichainContractList(
     walletAddress: string,
+    chains: Chain[] = defaultChains,
   ): Promise<ContractWithMetadata[]> {
     const contracts = await this.multiChainRegistry.getContractAddresses(
       walletAddress,
     );
 
+    const chainMap = chains.reduce((acc, chain) => {
+      acc[chain.chainId] = chain;
+      return acc;
+    }, {} as Record<number, Chain>);
+
     const sdkMap: Record<number, ThirdwebSDK> = {};
 
     return contracts.map(({ address, chainId }) => {
+      if (!chainMap[chainId]) {
+        // if we don't have the chain in our list of supported chains then we can't resolve the contract type regardless, don't even try to set up the SDK
+        return {
+          address,
+          chainId,
+          contractType: async () => "custom" as const,
+          metadata: async () => ({}),
```

Comment:
what happens in the contract detail for this when i click on it?

Reply:
it'll ask you to configure the chain